dispatches/models/fossil_case/ultra_supercritical_plant/ultra_supercritical_powerplant.py

```diff
@@ -0,0 +1,1424 @@
+    return m
+
+def _make_constraints(m):
+    # Define all model constraints except those included with the unit models
+    
+    #########   Boiler section   #########
+
+    # Following the reference DOE flowsheet, the outlet temperature for the
+    # boiler unit, i. e. boiler, reheater_1, & reheater_2 is fixed to 866.15 K
+    # Outlet temperature of boiler is set to 866.15 K
+    @m.fs.boiler.Constraint(m.fs.time)
+    def boiler_temperature_constraint(b, t):
+        return (b.control_volume.properties_out[t].temperature ==
+                866.15 * pyunits.K)
+
+    # Outlet temperature of reheater_1 is set to 866.15 K
+    @m.fs.reheater_1.Constraint(m.fs.time)
+    def reheater1_temperature_constraint(b, t):
+        return (b.control_volume.properties_out[t].temperature ==
+                866.15 * pyunits.K)
+
+    # Outlet temperature of reheater_2 is set to 866.15 K
+    @m.fs.reheater_2.Constraint(m.fs.time)
+    def reheater2_temperature_constraint(b, t):
+        return (b.control_volume.properties_out[t].temperature ==
+                866.15 * pyunits.K)
+
+    #########   Condenser section   #########
+    
+    # The inlet 'main' refers to the main steam coming from the turbine train
+    # Inlet 'bfpt' refers to the steam coming from the bolier feed pump turbine
+    # Inlet 'drain' refers to the condensed steam from the feed water heater 1
+    # Inlet 'makeup' refers to the make up water
+    # The outlet pressure of condenser mixer is equal to the minimum pressure
+    # Since the turbine (#11) outlet (or, mixer inlet 'main') pressure
+    # has the minimum pressure, the following constraint sets the outlet
+    # pressure of the condenser mixer to the pressure of the inlet 'main'
+    @m.fs.condenser_mix.Constraint(m.fs.time)
+    def mixer_pressure_constraint(b, t):
+        return b.main_state[t].pressure == b.mixed_state[t].pressure
+
+    # The outlet of condenser is assumed to be a saturated liquid
+    # The following condensate enthalpy at the outlet of condeser equal to
+    # that of a saturated liquid at that pressure
+    @m.fs.condenser.Constraint(m.fs.time)
+    def cond_vaporfrac_constraint(b, t):
+        return (
+            b.control_volume.properties_out[t].enth_mol
+            == b.control_volume.properties_out[t].enth_mol_sat_phase['Liq']
+        )
+
+    #########   Feed Water Heater section   #########
+
+    # The condensing steam is assumed to leave the FWH as saturated liquid
+    # Thus, each FWH is accompanied by 3 constraints, 2 for pressure drop
+    # and 1 for the enthalpy.
+
+    # The outlet pressure of fwh mixer is equal to the minimum pressure
+    # Since the pressure of mixer inlet 'steam' has the minimum pressure,
+    # the following constraint set the outlet pressure of fwh mixers
+    # to be same as the pressure of the inlet 'steam'
+    def fwhmixer_pressure_constraint(b, t):
+        return b.steam_state[t].pressure == b.mixed_state[t].pressure
```

Comment:
Have you tested the flowsheet with `"momentum_mixing_type": MomentumMixingType.minimize` instead of adding these constraints?

Reply:
I now did the testing and model solves fine, made the changes to the model accordingly